find/scripts/extract_download_logs.py

```diff
@@ -53,11 +56,54 @@ def send_notify(
     )
 
 
+def extract_ids_from_message(message: List[List[dict]]) -> List[str]:
+    """Extract user IDs from the message field in the log entries
+    args: message: List of log entries
+    returns: List of unique user IDs
+    """
+    account_ids = []
+    for log_entry in message:
+        for item in log_entry:
+            if item["field"] == "@message":
+                value = json.loads(item["value"])
+                account_ids.append(value["user_id"])
+    account_ids = list(set(account_ids))
+
+    return account_ids
+
+
+def fetch_user_data(user_ids: List[str]) -> dict:
+    """
+    Fetch user data from account-store API, using users' azure_ad_subject_id-s
+    args: user_ids: list of user_ids
+    returns: dict of user_id: email_address
+    """
+    responses = {}
+    for user_id in user_ids:
+        try:
+            response = requests.get(
+                f"{Config.ACCOUNT_STORE_API_HOST}/accounts",
+                headers={"Content-Type": "application/json"},
+                params={"azure_ad_subject_id": user_id},
+            )
+            response.raise_for_status()
+            response_data = response.json()
+            responses[user_id] = response_data.get("email_address")
```

Comment:
is `email_address` an optional field here? If it's not, should we do a hard lookup? If it is, should we add a default value if it's not present, like `no-email-address`?

Reply:
Yes, you're right, it's just that I most of the time use get instead of [], even if there's actually no chance of getting a key error. 
Email is not nullable, so I've changed it to a hard lookup.